tests/basic/volume_ops_test.go

```diff
@@ -1644,3 +1637,463 @@ var _ = Describe("{CreateFastpathVolumeRebootNode}", func() {
+			}
+
+			log.InfoD("snapshotScheduleName : %v for volume: %s", snapshotScheduleName, v.Name)
+			resp, err := storkops.Instance().GetSnapshotSchedule(snapshotScheduleName, appNamespace)
+			if err != nil {
+
+				return snapsMap, err
+			}
+
+			dash.VerifyFatal(len(resp.Status.Items) > 0, true, fmt.Sprintf("verify snapshots exists for %s", snapshotScheduleName))
+			for _, snapshotStatuses := range resp.Status.Items {
+				if len(snapshotStatuses) > 0 {
+					status := snapshotStatuses[len(snapshotStatuses)-1]
+					if status == nil {
+
+						return snapsMap, fmt.Errorf("SnapshotSchedule has an empty migration in it's most recent status")
+
+					}
+					log.Infof("Snapshot %s has status %v", status.Name, status.Status)
+
+					if status.Status == snapv1.VolumeSnapshotConditionError {
+
+						return snapsMap, fmt.Errorf("snapshot: %s failed. status: %v", status.Name, status.Status)
+					}
+					if status.Status != snapv1.VolumeSnapshotConditionPending {
+						snapData, err := Inst().S.GetSnapShotData(ctx, status.Name, appNamespace)
+
+						if err != nil {
+							return snapsMap, err
+						}
+
+						snapType := snapData.Spec.PortworxSnapshot.SnapshotType
+						log.Infof("Snapshot Type: %v", snapType)
+						if snapType != "cloud" {
+							err = &scheduler.ErrFailedToGetVolumeParameters{
+								App:   ctx.App,
+								Cause: fmt.Sprintf("Snapshot Type: %s does not match", snapType),
+							}
+							return snapsMap, err
+						}
+
+						snapID := snapData.Spec.PortworxSnapshot.SnapshotID
+						log.Infof("Snapshot ID: %v", snapID)
+						if snapData.Spec.VolumeSnapshotDataSource.PortworxSnapshot == nil ||
+							len(snapData.Spec.VolumeSnapshotDataSource.PortworxSnapshot.SnapshotID) == 0 {
+							err = &scheduler.ErrFailedToGetVolumeParameters{
+								App:   ctx.App,
+								Cause: fmt.Sprintf("volumesnapshotdata: %s does not have portworx volume source set", snapData.Metadata.Name),
+							}
+							return snapsMap, err
+						}
+
+					}
+					snapsMap[snapshotScheduleName] = status.Name
+
+				}
+			}
+
+		}
+	}
+	return snapsMap, nil
+}
+
+func trashcanRestore(volId, volName string) error {
+	log.InfoD("Restoring vol [%v] from trashcan", volId)
+	pxctlCmdFull := fmt.Sprintf("v r %s --trashcan %s", volName, volId)
+	output, err := Inst().V.GetPxctlCmdOutput(node.GetStorageNodes()[0], pxctlCmdFull)
+	if err != nil {
+		return err
+	}
+	log.Infof("output: %v", output)
+	if !strings.Contains(output, fmt.Sprintf("Successfully restored: %s", volName)) {
+		err = fmt.Errorf("volume %v, restore from trashcan failed, Err: %v", volId, output)
+		return err
+	}
+	return nil
+}
+
+func deletePXVolume(volName string) error {
+	delVol := func() (interface{}, bool, error) {
+		err := Inst().V.DeleteVolume(volName)
+		if err != nil {
+			return "", true, err
+		}
+		return nil, false, nil
+	}
+	_, err := task.DoRetryWithTimeout(delVol, time.Duration(60)*defaultCommandTimeout, 2*time.Minute)
+	return err
+}
```

Comment:
Are we ok to keep these functions outside common.go?

Reply:
ya.. these functions are currently used in this test only , if needed we can move them later